tests/test_mock_controller.py

```diff
@@ -43,85 +38,93 @@
 # Long timeout in case of a mock network issue
 SLOW_NETWORK_TIMEOUT = mtdome.MockMTDomeController.SLOW_NETWORK_SLEEP + 1.0
 
+INDEX_ITER = utils.index_generator()
 
-class MockTestCase(unittest.IsolatedAsyncioTestCase):
-    async def determine_current_tai(self) -> None:
-        pass
+
+class MockTestCase(tcpip.BaseOneClientServerTestCase):
+    server_class = mtdome.MockMTDomeController
 
     async def asyncSetUp(self) -> None:
-        self.ctrl = None
-        self.writer = None
-        port = 0
+        self.command_id: typing.Any = -1
```

Comment:
Note that there is nothing async in here. Perhaps setUp?

Please add a code comment explaining typing.Any (I think the point is to be able to specify invalid data types) and explain that valid data is an int.

Reply:
The `MockTestCase` inherits from `tcpip.BaseOneClientServerTestCase` which certainly is an async test case, so `asyncSetUp` needs to stay.

I added a comment to `self.command_id`.

Reply:
Actually `setUp` and `tearDown` work fine in async unit tests. unittest.IsolatedAsyncioTestCase (spelling?) *adds* `asyncSetUp` and `asyncTearDown`, but also runs the standard synchronous `setUp` and `tearDown`.

I have found one situation where I needed `asyncSetUp` when it's not obvious: constructing an object that needs a running event loop in the constructor. But I don't think this is such a case.

Reply:
That is really good to know!